Eval: ASL Classifiers

docequity opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, pelase note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑
### Eval name
ASL Classifiers

### Eval description

Classifiers are an integral part of American Sign Language (ASL) grammar. ASL classifiers are certain categories of hand shapes that are used to represent objects, actions, or locations. Given a description and an ASL classifier, the model is asked to determine whether the ASL translation of the description could be represented by the ASL classifier.

### What makes this a useful eval?

There are many evals for assessing text-based languages, but none for languages rooted in the physical environment. This task tests the model's ability to process the syntax of a non-textual language which is a useful entry into languages requiring multimodal input.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [x] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
  {"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"a walking person\" CL:2"}],"ideal":"YES"}
{"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"people standing in line\" CL:4"}],"ideal":"YES"}
{"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"a plane taking off\" CL:O"}],"ideal":"NO"}
{"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"a sailing boat\" CL:3"}],"ideal":"YES"}
{"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"a flying mosquito\" CL:F"}],"ideal":"YES"}
{"input":[{"role":"system","content":"You are an American Sign Language (ASL) interpreter. You will be prompted with a description in quotes followed by an ASL classifier. You want to see if you can sign the given description using the given ASL classifier. Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO."},{"role":"user","content":"\"a cluster of houses\" CL:I"}],"ideal":"NO"}
  ```
</details>

kzylam commented:
Thanks for the approval! Would love to get 32K access if possible since I already have GPT-4.

usama-openai reviewed (changes requested):
Sorry for the oversight. I would like to request a change.

I would recommend asking the model to reason before answering and using the `Includes` evaluation method here because such complex tasks are hard for the model to do zero-shot without a chance to reason through the steps. Asking the model to provide the chain of thought will give the model a fair chance to solve the problem, and a proper evaluation method can be used to evaluate the provided answer. You can also add instructions to provide output in a specific format and then use that format to write the ideal answer. For example, the model can be asked to enclose the final answer in square brackets, and the ideal answer can be formatted like `[Y]`. It'll give the model a proper chance to reason before answering, and the proper formatting will help in identifying the final answer using the `Includes` method.

docequity commented:
Thanks for your feedback. I have updated the evaluation method to use `Includes` and also updated the prompt to ask for the model's thought process. Invoking the thought process provided cool insights that the model understood what ASL classifiers were and had some decent ideas of how to apply them in a translation but still only had an accuracy of 0.55 with GPT-3.5-TURBO and an accuracy of 0.625 with GPT-4, which is similar to the original test runs.

usama-openai reviewed (changes requested):
Thanks for implementing the requested changes.

The input prompt looks a little confusing in the sense that first it says model to "go through thought process" and then it says to "Answer with either a YES or NO and provide no additional text.". You can change your prompt to something like: `Briefly go through your thought process for how the description could be or could not be represented by the classifier and write your final answer as YES or NO`.

docequity commented:
Thanks for the feedback. Had a brain fart between Match and Includes. I have updated the prompt.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

usama-openai commented:
You should see GPT-4 API access enabled in your account in the next few days.

docequity commented:
@usama-openai Since we already have GPT-4, is there any way we can get GPT-4-32k access instead? We have large data we'd like to do some tests on. If that is not possible, is there a chance you could grant GPT-4 access to the kzylam account that I accidentally commented with above? Thanks!
